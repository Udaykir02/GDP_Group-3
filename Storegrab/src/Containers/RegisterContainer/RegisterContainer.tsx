import React, { useEffect, useState } from 'react';
import { View, TextInput, Button, StyleSheet, Image, TouchableHighlight, Text, Alert } from 'react-native';
import axios from 'axios';
import createToBeImplementedStyle from "./RegisterContainerStyle";
import { useDispatch, useSelector } from 'react-redux';
import { registerRequest } from '../../../actions/userActions';
import { clearError } from '../../../reducers/users/slice';
import AppPageWrapper from '../../shared/AppPageWrapper';
import LoadingSpinner from '../../Components/shared/LoadingSpinner';

const RegisterContainer = ({ navigation }: any) => {
  const styles = createToBeImplementedStyle();
  const [username, setUsername] = useState('');
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const dispatch = useDispatch();
  const { error, loading } = useSelector((state:any)=> state.auth);
  const handleRegister = async () => {

      dispatch(registerRequest(username, email, password));

  };

  useEffect(()=>{
    // Display alert when error occurs
    if (error) {
        console.log(JSON.stringify(error))
        Alert.alert('Error', error, [{ text: 'OK' }]);
        dispatch(clearError())
    }
},[loading])

if(loading){
    return (<AppPageWrapper>
        <LoadingSpinner />
    </AppPageWrapper>)
}
  return (
    <View style={styles.container}>
      <View style={styles.imageContainer}>
        <Image
          source={require('../../Assets/icons/Storegrab.png')}
          style={styles.image}
        />
      </View>
      <View style={styles.formContainer}>
        <TextInput
          style={styles.input}
          placeholder="Email"
          value={email}
          onChangeText={setEmail}
          autoCapitalize={'none'}
        />
        <TextInput
          style={styles.input}
          placeholder="Username" // Add TextInput for username
          value={username}
          onChangeText={setUsername}
          autoCapitalize={'none'}
        />
        <TextInput
          style={styles.input}
          placeholder="Password"
          value={password}
          onChangeText={setPassword}
          secureTextEntry
          autoCapitalize={'none'}
        />
        <TouchableHighlight
          style={styles.button}
          onPress={handleRegister}
          underlayColor='#fff'>
          <Text style={[{ fontSize: 15 }, styles.loginText]}>Register</Text>
        </TouchableHighlight>
      </View>

    </View>
  );
};

export default RegisterContainer;
