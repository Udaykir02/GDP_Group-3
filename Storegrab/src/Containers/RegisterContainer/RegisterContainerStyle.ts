import { argonTheme } from '../../constants';
import { StyleSheet } from 'react-native';
export default function <C>() {
    return StyleSheet.create({
      container: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
        paddingHorizontal: 20,
      },
      title: {
        fontSize: 24,
        fontWeight: 'bold',
        marginBottom: 20,
      },
      input: {
        width: '100%',
        height: 40,
        borderColor: 'gray',
        borderWidth: 1,
        borderRadius: 5,
        paddingHorizontal: 10,
        marginBottom: 10,
      },
      button: {
        marginRight: 50,
        marginLeft: 50,
        marginTop: 20,
        paddingTop: 15,
        paddingBottom: 15,
        backgroundColor: argonTheme.COLORS.ERROR,
        borderRadius: 10,
        borderWidth: 1,
        borderColor: '#fff',
      },
      buttonText: {
        color: 'white',
        fontSize: 16,
        fontWeight: 'bold',
      },
      loginText: {
        color: '#fff',
        textAlign: 'center',
        fontSize: 16
      },
      image: {
        width: 200,
        height: 220,
        justifyContent: 'center',
        alignSelf: 'center'
      },
      imageContainer: { flex: 1, justifyContent: 'center' },
      formContainer: { flex: 1, width: '100%', padding: 20 }
    })
}