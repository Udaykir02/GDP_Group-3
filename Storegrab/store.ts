import { configureStore } from '@reduxjs/toolkit';
import createSagaMiddleware from 'redux-saga';
import rootReducers from './reducers'; // Assuming rootReducers is defined in a separate file
import rootSaga from './sagas/rootSaga'; // Assuming rootSaga is defined in a separate file

// Create saga middleware instance
const sagaMiddleware = createSagaMiddleware();

// Create the Redux store with middleware and reducers
const store = configureStore({
  reducer: rootReducers,
  middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(sagaMiddleware),
});

// Run the root saga
sagaMiddleware.run(rootSaga);

export default store;
