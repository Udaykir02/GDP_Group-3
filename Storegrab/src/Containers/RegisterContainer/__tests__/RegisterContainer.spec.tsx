import { Provider } from 'react-redux'
import store from '../../../../store'
import render, {
    RenderResult,
} from '@testing-library/react-native/build/render'


import { StripeProvider } from '@stripe/stripe-react-native';
import { PaperProvider } from 'react-native-paper';
import AppThemeProvider from '../../../core/AppThemeProvider';
import RegisterContainer from '../RegisterContainer';

jest.mock('@react-navigation/native', () => {
    return {
        ...jest.requireActual('@react-navigation/native'),
        useNavigation: () => ({
            navigate: jest.fn(),
        }),
    };
});

const renderStartupContainerPage = () => {
    return render(
            <Provider store={store}>
                <PaperProvider>
                    <AppThemeProvider>
                        <RegisterContainer />
                    </AppThemeProvider>
                </PaperProvider>
            </Provider>
    )
}
let page: RenderResult

describe('Prodcard', () => {
    beforeEach(() => {
        jest.useFakeTimers();
        page = renderStartupContainerPage()
    })
    afterEach(() => {
        jest.runOnlyPendingTimers();
        jest.useRealTimers();
    });
    it('should render correctly', function () {
        expect(page.toJSON()).toMatchSnapshot()
    })

});