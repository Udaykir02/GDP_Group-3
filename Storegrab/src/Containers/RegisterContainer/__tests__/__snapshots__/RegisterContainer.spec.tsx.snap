// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prodcard should render correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    collapsable={false}
    pointerEvents="box-none"
    style={
      {
        "flex": 1,
      }
    }
  >
    <View
      style={
        [
          {
            "flex": 1,
          },
          undefined,
        ]
      }
    >
      <View
        collapsable={false}
        pointerEvents="box-none"
        style={
          {
            "flex": 1,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
              "paddingHorizontal": 20,
            }
          }
        >
          <View
            style={
              {
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Image
              source={
                {
                  "testUri": "../../../src/Assets/icons/Storegrab.png",
                }
              }
              style={
                {
                  "alignSelf": "center",
                  "height": 220,
                  "justifyContent": "center",
                  "width": 200,
                }
              }
            />
          </View>
          <View
            style={
              {
                "flex": 1,
                "padding": 20,
                "width": "100%",
              }
            }
          >
            <TextInput
              autoCapitalize="none"
              onChangeText={[Function]}
              placeholder="Email"
              style={
                {
                  "borderColor": "gray",
                  "borderRadius": 5,
                  "borderWidth": 1,
                  "height": 40,
                  "marginBottom": 10,
                  "paddingHorizontal": 10,
                  "width": "100%",
                }
              }
              value=""
            />
            <TextInput
              autoCapitalize="none"
              onChangeText={[Function]}
              placeholder="Username"
              style={
                {
                  "borderColor": "gray",
                  "borderRadius": 5,
                  "borderWidth": 1,
                  "height": 40,
                  "marginBottom": 10,
                  "paddingHorizontal": 10,
                  "width": "100%",
                }
              }
              value=""
            />
            <TextInput
              autoCapitalize="none"
              onChangeText={[Function]}
              placeholder="Password"
              secureTextEntry={true}
              style={
                {
                  "borderColor": "gray",
                  "borderRadius": 5,
                  "borderWidth": 1,
                  "height": 40,
                  "marginBottom": 10,
                  "paddingHorizontal": 10,
                  "width": "100%",
                }
              }
              value=""
            />
            <View
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "backgroundColor": "#4CAF50",
                  "borderColor": "#fff",
                  "borderRadius": 10,
                  "borderWidth": 1,
                  "marginLeft": 50,
                  "marginRight": 50,
                  "marginTop": 20,
                  "paddingBottom": 15,
                  "paddingTop": 15,
                }
              }
            >
              <Text
                style={
                  [
                    {
                      "fontSize": 15,
                    },
                    {
                      "color": "#fff",
                      "fontSize": 16,
                      "textAlign": "center",
                    },
                  ]
                }
              >
                Register
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
